src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerDifferenceService.kt

```diff
@@ -146,13 +140,13 @@ class PrisonerDifferenceService(
     prisoner: Prisoner,
   ) {
     previousPrisonerSnapshot?.also {
-      val differences = getDifferencesByCategory(it, prisoner)
+      val differences = reportDiffTelemetryDetails(previousPrisonerSnapshot, prisoner)
       if (differences.isNotEmpty()) {
         telemetryClient.trackEvent(
           "POSPrisonerDifferenceReported",
           mapOf(
             "nomsNumber" to previousPrisonerSnapshot.prisonerNumber,
-            "categoriesChanged" to differences.keys.map { it.name }.toList().sorted().toString(),
+            "differences" to differences.toString(),
```

Comment:
report all the differences rather than just the categories since we are going to lose the information after re-index.

Reply:
I did this originally but it was pointed out that this could publish Personally Identifiable Information to App Insights. Which is bad apparently. Hence I switched to categories.

Reply:
Could we just publish telemetry containing the field names but not the values?

Reply:
fixed - differences now to database instead